### PR TITLE
Added type information to tests

### PR DESCRIPTION
**Pull request by ssbarnea**
Partial-Fix: #893

### CODE REVIEW

File: lib/ansiblelint/runner.py [hunk 1]
```diff
@@ -23,9 +23,9 @@ def __init__(
             self,
             rules: "RulesCollection",
             playbook: str,
-            tags: FrozenSet[Any],
-            skip_list: FrozenSet[Any],
-            exclude_paths: List[str],
+            tags: FrozenSet[Any] = frozenset(),
+            skip_list: Optional[FrozenSet[Any]] = frozenset(),
+            exclude_paths: List[str] = [],
```
Comment (webknjaz) [inline lib/ansiblelint/runner.py:28]: Lists are mutable. You shouldn't use them as method defaults.

File: lib/ansiblelint/rules/__init__.py [hunk 2]
```diff
@@ -190,7 +190,7 @@ def __init__(self, rulesdirs=None) -> None:
             self.extend(load_plugins(rulesdir))
         self.rules = sorted(self.rules, key=lambda r: r.id)
 
-    def register(self, obj):
+    def register(self, obj: AnsibleLintRule):
```
Comment (webknjaz) [inline lib/ansiblelint/rules/__init__.py:193]: How is this related to tests?

**Review by webknjaz — CHANGES_REQUESTED** [review]

File: lib/ansiblelint/runner.py (hunk 1 above)
Comment (ssbarnea) [reply lib/ansiblelint/runner.py:28]: Using an empty list is the only way to make the parameter optional without changing the API. I do not see any problem with having a list passed. 

If you want to change exclude_paths to be something else I invite you to make PR that does this, outside of this change.

File: lib/ansiblelint/rules/__init__.py (hunk 2 above)
Comment (ssbarnea) [reply lib/ansiblelint/rules/__init__.py:193]: Yes it does. I would really recommend you to try adding some typing information yourself on ansible-lint, because that is something you will realise right away: adding type to one method does create a long chain of dependencies.

Here is what happens if I remove it:
```
test/TestWithSkipTagId.py:10: error: Call to untyped function "register" in typed context
test/TestMetaMainHasInfo.py:61: error: Call to untyped function "register" in typed context
test/TestUseHandlerRatherThanWhenChanged.py:77: error: Call to untyped function "register" in typed context
test/TestComparisonToEmptyString.py:28: error: Call to untyped function "register" in typed context
test/TestSudoRule.py:48: error: Call to untyped function "register" in typed context
test/TestLineTooLong.py:17: error: Call to untyped function "register" in typed context
test/TestVariableHasSpaces.py:47: error: Call to untyped function "register" in typed context
test/TestMetaChangeFromDefault.py:19: error: Call to untyped function "register" in typed context
test/TestMetaVideoLinks.py:23: error: Call to untyped function "register" in typed context
test/TestTaskNoLocalAction.py:17: error: Call to untyped function "register" in typed context
test/TestDeprecatedModule.py:19: error: Call to untyped function "register" in typed context
test/TestShellWithoutPipefail.py:73: error: Call to untyped function "register" in typed context
test/TestEnvVarsInCommand.py:79: error: Call to untyped function "register" in typed context
test/TestRoleRelativePath.py:41: error: Call to untyped function "register" in typed context
test/TestMetaTagValid.py:17: error: Call to untyped function "register" in typed context
test/TestComparisonToLiteralBool.py:46: error: Call to untyped function "register" in typed context
Found 16 errors in 16 files (checked 117 source files)
```
